Clarify and update ActiveRecord Callbacks documentation

olivierlacan opened the pull request:
I let a previous PR (#37731) get stale, this a redo with the changes suggested by @p8.

---

Mainly these changes aim to make this documentation a bit more ESL (English
as a second language friendly).

Here are some examples of the things I changed:
- replace the constant use of `get <verb>` with `is <verb>`
- replace $10 words like `alteration` with clearer $1 words like `change`
- `give you immense power` -> `offer you a lot of control`
- missing `<code>` wrapping to denote method/variable references and SQL
statements like `COMMIT` and `ROLLBACK`.
- clarify the ordering callback example specifically to explain what children is
(an association with records)

This is minimal stuff but I think it improves the clarity and legibility of this
extremely important piece of documentation. I'm actually surprised that it's not
linked to from the [Callbacks Guide][1]

[1]: https://guides.rubyonrails.org/active_record_callbacks.html

olivierlacan committed:
Clarify documentation on ActiveRecord Callbacks

Mainly these changes aim to make this documentation a bit more ESL (English
as a second language friendly).

Here are some examples of the things I changed:
- replace the constant use of `get <verb>` with `is <verb>`
- replace $10 words like `alteration` with clearer $1 words like `change`
- `give you immense power` -> `offer you a lot of control`
- missing `<code>` wrapping to denote method/variable references and SQL
statements like `COMMIT` and `ROLLBACK`.
- clarify the ordering callback example specifically to explain what children is
(an association with records)

This is minimal stuff but I think it improves the clarity and legibility of this
extremely important piece of documentation. I'm actually surprised that it's not
linked to from the [Callbacks Guide][1]

[ci skip]

[1]: https://guides.rubyonrails.org/active_record_callbacks.html

p8 commented:
Ah sorry, I've picked up the three/four myself  :) #38553

olivierlacan commented:
@p8 Thief! It's cool, I'll rebase when you merge.

p8 commented:
@olivierlacan my PR got merged 😄

BKSpurgeon commented:
I think you can add ` [ci skip]` to the PR title for pure documentation changes to skip the ci build.

Schwad reviewed (approved):
This looks great. I think changes like this will help all devs parsing the documentation (as well as ESL devs). Nice!

olivierlacan commented:
> I think you can add ` [ci skip]` to the PR title for pure documentation changes to skip the ci build.

Added it to the commit message, I think that should work too. But I originally didn't, thanks for the reminder.

Just rebased master so the changes from @p8 are included.
